### PR TITLE
feat: aws-lambda feature

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/FeatureUtils.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.starter.application.Project;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.template.RockerTemplate;
+
+public final class FeatureUtils {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/FeatureUtils.java:23]: Do we need a new class here, can these methods not just go into `GeneratorContext`? Please delete this.

File: starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.starter.application.Project;
+
+public interface SourceTemplateProvider {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java:21]: Why are we exposing this interface? These methods are only used internally within a feature. Please delete this.

File: starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ */
+package io.micronaut.starter.feature;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.starter.application.Project;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.options.Language;
+import io.micronaut.starter.options.TestFramework;
+
+public interface TestTemplateProvider {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java:24]: Same thing why the need for an interface? These are not going to be called outside the feature. Please delete this.

File: starter-core/src/main/java/io/micronaut/starter/feature/FeatureUtils.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/FeatureUtils.java]: I've removed this class and added the methods into `GeneratorContext`

**Comment by sdelamo** [comment]
Regarding `TestTemplateProvider` and `SourceTemplateProvider`  I am leveraging these interfaces to create features such as the `AwsLambda`. 

I have pushed the AWS Lambda feature upstream so that you can see how I use them. 

Do you want me to remove them?

File: starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java:21]: I am using this in AwsLambda Feature and also in AzureFunction feature and I think they will be used in many other features. Do you still want me to remove them?

File: starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java (hunk 3 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java:24]: I am using this in AwsLambda Feature and also in AzureFunction feature and I think they will be used in many other features. Do you still want me to remove them?

**Comment by graemerocher** [comment]
They are used internally within a feature, there is no need for them, they could just as easily be super class methods without needing an interface

**Comment by sdelamo** [comment]
I have removed the interfaces and created extra parameters in the methods. 
https://github.com/micronaut-projects/micronaut-starter/pull/117/commits/9b18081304e5832559d00c3aa5ca717731228e2c

File: starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/SourceTemplateProvider.java]: I have deleted them

File: starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java (hunk 3 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/TestTemplateProvider.java]: I've deleted them.

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaBookMicronautRequestHandlerGroovyJunit.rocker.raw [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+package @project.getPackageName()
+}
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class BookMicronautRequestHandlerTest {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaBookMicronautRequestHandlerGroovyJunit.rocker.raw:17]: Rename to `BookRequestHandlerTest`

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaBookMicronautRequestHandlerJava.rocker.raw [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+@if (project.getPackageName() != null) {
+package @project.getPackageName()
+}
+
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.function.aws.MicronautRequestHandler;
+import java.util.UUID;
+
+@@Introspected
+public class BookMicronautRequestHandler extends MicronautRequestHandler<Book, BookSaved> {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaBookMicronautRequestHandlerJava.rocker.raw:16]: Rename to `BookRequestHandler`

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import io.micronaut.starter.feature.function.awslambda.template.awsLambdaBookMicronautRequestHandlerKotlinTest;
+import io.micronaut.starter.feature.function.awslambda.template.awsLambdaBookMicronautRequestHandlerSpock;
+import io.micronaut.starter.feature.function.awslambda.template.awsLambdaBookSavedGroovy;
+import io.micronaut.starter.feature.function.awslambda.template.awsLambdaBookSavedJava;
+import io.micronaut.starter.feature.function.awslambda.template.awsLambdaBookSavedKotlin;
+
+import javax.inject.Singleton;
+
+@Singleton
+public class AwsLambda implements Feature {
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java:40]: Should implement `DefaultFeature` as AWS Lambda will be the default for `create-function`

**Pull request merged**